compiler/src/dotty/tools/dotc/quoted/PickledQuotes.scala

```diff
@@ -248,23 +249,39 @@ object PickledQuotes {
           case pickled: String => TastyString.unpickle(pickled)
           case pickled: List[String] => TastyString.unpickle(pickled)
 
-        quotePickling.println(s"**** unpickling quote from TASTY\n${TastyPrinter.showContents(bytes, ctx.settings.color.value == "never")}")
+        val unpicklingContext =
+          if ctx.owner.isClass then
+            // When a quote is unpickled with a Quotes context that that has a class `spliceOwner`
+            // we need to use a dummy owner to unpickle it. Otherwise any definitions defined
+            // in the quoted block would be accidentally entered in the class.
+            // This owner is replaced with the correct owner when at the splice site.
```

Comment:
Is that in `callMacro`? Maybe refer to it explicitly, then it's clearer where the owner gets replaced.

Reply:
It is in this same file in `quotedExprToTree` and `quotedTypeToTree`. I added a reference in the comment and explained a bit more.